File: exercise/leetcode/0049/0049.py
from typing import List

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        tb = {}
        for i, s in enumerate(strs):
            tmp = list(s)
            tmp.sort()
            tmp = ''.join(tmp)
            if tmp in tb:
                tb[tmp].append(i)
            else:
                tb[tmp] = [i]
        rst = []
        for k, lst in tb.items():
            tmp = []
            for i in lst:
                tmp.append(strs[i])
            rst.append(tmp)
        return rst

File: exercise/leetcode/0049/0049.cc
#include <iostream>
#include <vector>
#include <string>
#include <unordered_map>
#include <algorithm>
using namespace std;

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<int>> tb;

        for (int i = 0; i < strs.size(); ++i) {
            string tmp = strs[i];
            sort(tmp.begin(), tmp.end());
            tb[tmp].push_back(i);
        }

        vector<vector<string>> rst;
        for (auto it = tb.begin(); it != tb.end(); ++it) {
            vector<string> tmp;
            vector<int>& info = it->second;
            for (int i = 0; i < info.size(); ++i) {
                tmp.push_back(strs[info[i]]);
            }
            rst.emplace_back(move(tmp));
        }

        return rst;

    }
};